compendium/DeclarativeServices/test/gtest/TestAsyncWorkService.cpp

```diff
@@ -312,10 +287,9 @@ namespace test
 
     INSTANTIATE_TEST_SUITE_P(AsyncWorkServiceEndToEndParameterized,
                              TestAsyncWorkServiceEndToEnd,
-                             testing::Values(std::make_shared<AsyncWorkServiceInline>(),
```

Comment:
`AsyncWorkServiceInline` was meant to simulate essentially no thread pool; the work is done inline on the current thread. I think there is still value in making sure this use case works.

Reply:
I don't think this is always possible. I ran into a deadlock grabbing a lock where the work is done on the same thread. Essentially if you perform a configChangeState which causes a service to be satisfied, that eventually causes a listener in configAdmin to be notified and notify others of this serviceUpdated... So I don't think this is actually possible on a single thread. 

In:
configuration->Update() (Client)
eventually calls post (from perform Async in notifyConfigurationUpdated) (CA)
post executes the function (AsyncWS)
configurationEvent (DS)
NotifyAllListeners (DS)
notify (DS)
configChangedState (DS)
Register()
calls mgr.RegisterService()
RegistrationManager -> bundleContext.RegisterService()
ServiceRegistry calls serviceChanged()
eventually gets to tracker -- tracks reference
bundleAbstractTracked::Track
eventually gets back to CA perform async locking on the futuresMutex... 

Full stack trace here:

`libpthread.so.0!__lll_lock_wait(int * futex, int * futex@entry, int private) (lowlevellock.c:52)
libpthread.so.0!__GI___pthread_mutex_lock(pthread_mutex_t * mutex) (pthread_mutex_lock.c:80)
libConfigurationAdmind.so!__gthread_mutex_lock(__gthread_mutex_t * __mutex) (/usr/include/x86_64-linux-gnu/c++/10/bits/gthr-default.h:749)
libConfigurationAdmind.so!std::mutex::lock(class std::mutex * const this) (/usr/include/c++/10/bits/std_mutex.h:100)
libConfigurationAdmind.so!std::lock_guard<std::mutex>::lock_guard(class std::lock_guard<std::mutex> * const this, std::lock_guard<std::mutex>::mutex_type & __m) (/usr/include/c++/10/bits/std_mutex.h:159)
libConfigurationAdmind.so!cppmicroservices::cmimpl::ConfigurationAdminImpl::PerformAsync<cppmicroservices::cmimpl::ConfigurationAdminImpl::AddingService(const cppmicroservices::ServiceReference<cppmicroservices::service::cm::ManagedService>&)::<lambda()> >(struct {...} &&)(class cppmicroservices::cmimpl::ConfigurationAdminImpl * const this, struct {...} && f) (/home/tcormack/dev/AsyncWorkDeadlock/CppMicroServices/compendium/ConfigurationAdmin/src/ConfigurationAdminImpl.cpp:1037)
libConfigurationAdmind.so!cppmicroservices::cmimpl::ConfigurationAdminImpl::AddingService(class cppmicroservices::cmimpl::ConfigurationAdminImpl * const this, const class cppmicroservices::ServiceReference<cppmicroservices::service::cm::ManagedService> & reference) (/home/tcormack/dev/AsyncWorkDeadlock/CppMicroServices/compendium/ConfigurationAdmin/src/ConfigurationAdminImpl.cpp:856)
libConfigurationAdmind.so!cppmicroservices::detail::TrackedService<cppmicroservices::service::cm::ManagedService, cppmicroservices::ServiceTrackerCustomizer<cppmicroservices::service::cm::ManagedService, cppmicroservices::cmimpl::TrackedServiceWrapper<cppmicroservices::service::cm::ManagedService> >::TypeTraits>::CustomizerAdding(class cppmicroservices::detail::TrackedService<cppmicroservices::service::cm::ManagedService, cppmicroservices::ServiceTrackerCustomizer<cppmicroservices::service::cm::ManagedService, cppmicroservices::cmimpl::TrackedServiceWrapper<cppmicroservices::service::cm::ManagedService> >::TypeTraits> * const this, class cppmicroservices::ServiceReference<cppmicroservices::service::cm::ManagedService> item) (/home/tcormack/dev/AsyncWorkDeadlock/CppMicroServices/framework/include/cppmicroservices/detail/TrackedService.hpp:143)
libConfigurationAdmind.so!cppmicroservices::detail::BundleAbstractTracked<cppmicroservices::ServiceReference<cppmicroservices::service::cm::ManagedService>, std::shared_ptr<cppmicroservices::cmimpl::TrackedServiceWrapper<cppmicroservices::service::cm::ManagedService> >, cppmicroservices::ServiceEvent>::TrackAdding(class cppmicroservices::detail::BundleAbstractTracked<cppmicroservices::ServiceReference<cppmicroservices::service::cm::ManagedService>, std::shared_ptr<cppmicroservices::cmimpl::TrackedServiceWrapper<cppmicroservices::service::cm::ManagedService> >, cppmicroservices::ServiceEvent> * const this, class cppmicroservices::ServiceReference<cppmicroservices::service::cm::ManagedService> item, class cppmicroservices::ServiceEvent related) (/home/tcormack/dev/AsyncWorkDeadlock/CppMicroServices/framework/include/cppmicroservices/detail/BundleAbstractTracked.hpp:303)
libConfigurationAdmind.so!cppmicroservices::detail::BundleAbstractTracked<cppmicroservices::ServiceReference<cppmicroservices::service::cm::ManagedService>, std::shared_ptr<cppmicroservices::cmimpl::TrackedServiceWrapper<cppmicroservices::service::cm::ManagedService> >, cppmicroservices::ServiceEvent>::Track(class cppmicroservices::detail::BundleAbstractTracked<cppmicroservices::ServiceReference<cppmicroservices::service::cm::ManagedService>, std::shared_ptr<cppmicroservices::cmimpl::TrackedServiceWrapper<cppmicroservices::service::cm::ManagedService> >, cppmicroservices::ServiceEvent> * const this, class cppmicroservices::ServiceReference<cppmicroservices::service::cm::ManagedService> item, class cppmicroservices::ServiceEvent related) (/home/tcormack/dev/AsyncWorkDeadlock/CppMicroServices/framework/include/cppmicroservices/detail/BundleAbstractTracked.hpp:141)
libConfigurationAdmind.so!cppmicroservices::detail::TrackedService<cppmicroservices::service::cm::ManagedService, cppmicroservices::ServiceTrackerCustomizer<cppmicroservices::service::cm::ManagedService, cppmicroservices::cmimpl::TrackedServiceWrapper<cppmicroservices::service::cm::ManagedService> >::TypeTraits>::ServiceChanged(class cppmicroservices::detail::TrackedService<cppmicroservices::service::cm::ManagedService, cppmicroservices::ServiceTrackerCustomizer<cppmicroservices::service::cm::ManagedService, cppmicroservices::cmimpl::TrackedServiceWrapper<cppmicroservices::service::cm::ManagedService> >::TypeTraits> * const this, const class cppmicroservices::ServiceEvent & event) (/home/tcormack/dev/AsyncWorkDeadlock/CppMicroServices/framework/include/cppmicroservices/detail/TrackedService.hpp:93)
libConfigurationAdmind.so!std::__invoke_impl<void, void (cppmicroservices::detail::TrackedService<cppmicroservices::service::cm::ManagedService, cppmicroservices::ServiceTrackerCustomizer<cppmicroservices::service::cm::ManagedService, cppmicroservices::cmimpl::TrackedServiceWrapper<cppmicroservices::service::cm::ManagedService> >::TypeTraits>::*&)(cppmicroservices::ServiceEvent const&), std::shared_ptr<cppmicroservices::detail::TrackedService<cppmicroservices::service::cm::ManagedService, cppmicroservices::ServiceTrackerCustomizer<cppmicroservices::service::cm::ManagedService, cppmicroservices::cmimpl::TrackedServiceWrapper<cppmicroservices::service::cm::ManagedService> >::TypeTraits> >&, cppmicroservices::ServiceEvent const&>(void (cppmicroservices::detail::TrackedService<cppmicroservices::service::cm::ManagedService, cppmicroservices::ServiceTrackerCustomizer<cppmicroservices::service::cm::ManagedService, cppmicroservices::cmimpl::TrackedServiceWrapper<cppmicroservices::service::cm::ManagedService> >::TypeTraits>::*&)(class cppmicroservices::detail::TrackedService<cppmicroservices::service::cm::ManagedService, cppmicroservices::ServiceTrackerCustomizer<cppmicroservices::service::cm::ManagedService, cppmicroservices::cmimpl::TrackedServiceWrapper<cppmicroservices::service::cm::ManagedService> >::TypeTraits> * const, const class cppmicroservices::ServiceEvent &) __f, class std::shared_ptr<cppmicroservices::detail::TrackedService<cppmicroservices::service::cm::ManagedService, cppmicroservices::ServiceTrackerCustomizer<cppmicroservices::service::cm::ManagedService, cppmicroservices::cmimpl::TrackedServiceWrapper<cppmicroservices::service::cm::ManagedService> >::TypeTraits> > & __t) (/usr/include/c++/10/bits/invoke.h:73)
libConfigurationAdmind.so!std::__invoke<void (cppmicroservices::detail::TrackedService<cppmicroservices::service::cm::ManagedService, cppmicroservices::ServiceTrackerCustomizer<cppmicroservices::service::cm::ManagedService, cppmicroservices::cmimpl::TrackedServiceWrapper<cppmicroservices::service::cm::ManagedService> >::TypeTraits>::*&)(cppmicroservices::ServiceEvent const&), std::shared_ptr<cppmicroservices::detail::TrackedService<cppmicroservices::service::cm::ManagedService, cppmicroservices::ServiceTrackerCustomizer<cppmicroservices::service::cm::ManagedService, cppmicroservices::cmimpl::TrackedServiceWrapper<cppmicroservices::service::cm::ManagedService> >::TypeTraits> >&, cppmicroservices::ServiceEvent const&>(void (cppmicroservices::detail::TrackedService<cppmicroservices::service::cm::ManagedService, cppmicroservices::ServiceTrackerCustomizer<cppmicroservices::service::cm::ManagedService, cppmicroservices::cmimpl::TrackedServiceWrapper<cppmicroservices::service::cm::ManagedService> >::TypeTraits>::*&)(class cppmicroservices::detail::TrackedService<cppmicroservices::service::cm::ManagedService, cppmicroservices::ServiceTrackerCustomizer<cppmicroservices::service::cm::ManagedService, cppmicroservices::cmimpl::TrackedServiceWrapper<cppmicroservices::service::cm::ManagedService> >::TypeTraits> * const, const class cppmicroservices::ServiceEvent &) __fn) (/usr/include/c++/10/bits/invoke.h:95)
libConfigurationAdmind.so!std::_Bind<void (cppmicroservices::detail::TrackedService<cppmicroservices::service::cm::ManagedService, cppmicroservices::ServiceTrackerCustomizer<cppmicroservices::service::cm::ManagedService, cppmicroservices::cmimpl::TrackedServiceWrapper<cppmicroservices::service::cm::ManagedService> >::TypeTraits>::*(std::shared_ptr<cppmicroservices::detail::TrackedService<cppmicroservices::service::cm::ManagedService, cppmicroservices::ServiceTrackerCustomizer<cppmicroservices::service::cm::ManagedService, cppmicroservices::cmimpl::TrackedServiceWrapper<cppmicroservices::service::cm::ManagedService> >::TypeTraits> >, std::_Placeholder<1>))(cppmicroservices::ServiceEvent const&)>::__call<void, cppmicroservices::ServiceEvent const&, 0ul, 1ul>(std::tuple<cppmicroservices::ServiceEvent const&>&&, std::_Index_tuple<0ul, 1ul>)(struct std::_Bind<void (cppmicroservices::detail::TrackedService<cppmicroservices::service::cm::ManagedService, cppmicroservices::ServiceTrackerCustomizer<cppmicroservices::service::cm::ManagedService, cppmicroservices::cmimpl::TrackedServiceWrapper<cppmicroservices::service::cm::ManagedService> >::TypeTraits>::*(std::shared_ptr<cppmicroservices::detail::TrackedService<cppmicroservices::service::cm::ManagedService, cppmicroservices::ServiceTrackerCustomizer<cppmicroservices::service::cm::ManagedService, cppmicroservices::cmimpl::TrackedServiceWrapper<cppmicroservices::service::cm::ManagedService> >::TypeTraits> >, std::_Placeholder<1>))(const cppmicroservices::ServiceEvent&)> * const this, class std::tuple<cppmicroservices::ServiceEvent const&> && __args) (/usr/include/c++/10/functional:416)
libConfigurationAdmind.so!std::_Bind<void (cppmicroservices::detail::TrackedService<cppmicroservices::service::cm::ManagedService, cppmicroservices::ServiceTrackerCustomizer<cppmicroservices::service::cm::ManagedService, cppmicroservices::cmimpl::TrackedServiceWrapper<cppmicroservices::service::cm::ManagedService> >::TypeTraits>::*(std::shared_ptr<cppmicroservices::detail::TrackedService<cppmicroservices::service::cm::ManagedService, cppmicroservices::ServiceTrackerCustomizer<cppmicroservices::service::cm::ManagedService, cppmicroservices::cmimpl::TrackedServiceWrapper<cppmicroservices::service::cm::ManagedService> >::TypeTraits> >, std::_Placeholder<1>))(cppmicroservices::ServiceEvent const&)>::operator()<cppmicroservices::ServiceEvent const&, void>(cppmicroservices::ServiceEvent const&)(struct std::_Bind<void (cppmicroservices::detail::TrackedService<cppmicroservices::service::cm::ManagedService, cppmicroservices::ServiceTrackerCustomizer<cppmicroservices::service::cm::ManagedService, cppmicroservices::cmimpl::TrackedServiceWrapper<cppmicroservices::service::cm::ManagedService> >::TypeTraits>::*(std::shared_ptr<cppmicroservices::detail::TrackedService<cppmicroservices::service::cm::ManagedService, cppmicroservices::ServiceTrackerCustomizer<cppmicroservices::service::cm::ManagedService, cppmicroservices::cmimpl::TrackedServiceWrapper<cppmicroservices::service::cm::ManagedService> >::TypeTraits> >, std::_Placeholder<1>))(const cppmicroservices::ServiceEvent&)> * const this) (/usr/include/c++/10/functional:499)
libConfigurationAdmind.so!std::__invoke_impl<void, std::_Bind<void (cppmicroservices::detail::TrackedService<cppmicroservices::service::cm::ManagedService, cppmicroservices::ServiceTrackerCustomizer<cppmicroservices::service::cm::ManagedService, cppmicroservices::cmimpl::TrackedServiceWrapper<cppmicroservices::service::cm::ManagedService> >::TypeTraits>::*(std::shared_ptr<cppmicroservices::detail::TrackedService<cppmicroservices::service::cm::ManagedService, cppmicroservices::ServiceTrackerCustomizer<cppmicroservices::service::cm::ManagedService, cppmicroservices::cmimpl::TrackedServiceWrapper<cppmicroservices::service::cm::ManagedService> >::TypeTraits> >, std::_Placeholder<1>))(cppmicroservices::ServiceEvent const&)>&, cppmicroservices::ServiceEvent const&>(std::__invoke_other, std::_Bind<void (cppmicroservices::detail::TrackedService<cppmicroservices::service::cm::ManagedService, cppmicroservices::ServiceTrackerCustomizer<cppmicroservices::service::cm::ManagedService, cppmicroservices::cmimpl::TrackedServiceWrapper<cppmicroservices::service::cm::ManagedService> >::TypeTraits>::*(std::shared_ptr<cppmicroservices::detail::TrackedService<cppmicroservices::service::cm::ManagedService, cppmicroservices::ServiceTrackerCustomizer<cppmicroservices::service::cm::ManagedService, cppmicroservices::cmimpl::TrackedServiceWrapper<cppmicroservices::service::cm::ManagedService> >::TypeTraits> >, std::_Placeholder<1>))(cppmicroservices::ServiceEvent const&)>&, cppmicroservices::ServiceEvent const&)(struct std::_Bind<void (cppmicroservices::detail::TrackedService<cppmicroservices::service::cm::ManagedService, cppmicroservices::ServiceTrackerCustomizer<cppmicroservices::service::cm::ManagedService, cppmicroservices::cmimpl::TrackedServiceWrapper<cppmicroservices::service::cm::ManagedService> >::TypeTraits>::*(std::shared_ptr<cppmicroservices::detail::TrackedService<cppmicroservices::service::cm::ManagedService, cppmicroservices::ServiceTrackerCustomizer<cppmicroservices::service::cm::ManagedService, cppmicroservices::cmimpl::TrackedServiceWrapper<cppmicroservices::service::cm::ManagedService> >::TypeTraits> >, std::_Placeholder<1>))(const cppmicroservices::ServiceEvent&)> & __f) (/usr/include/c++/10/bits/invoke.h:60)
libConfigurationAdmind.so!std::__invoke_r<void, std::_Bind<void (cppmicroservices::detail::TrackedService<cppmicroservices::service::cm::ManagedService, cppmicroservices::ServiceTrackerCustomizer<cppmicroservices::service::cm::ManagedService, cppmicroservices::cmimpl::TrackedServiceWrapper<cppmicroservices::service::cm::ManagedService> >::TypeTraits>::*(std::shared_ptr<cppmicroservices::detail::TrackedService<cppmicroservices::service::cm::ManagedService, cppmicroservices::ServiceTrackerCustomizer<cppmicroservices::service::cm::ManagedService, cppmicroservices::cmimpl::TrackedServiceWrapper<cppmicroservices::service::cm::ManagedService> >::TypeTraits> >, std::_Placeholder<1>))(cppmicroservices::ServiceEvent const&)>&, cppmicroservices::ServiceEvent const&>(std::_Bind<void (cppmicroservices::detail::TrackedService<cppmicroservices::service::cm::ManagedService, cppmicroservices::ServiceTrackerCustomizer<cppmicroservices::service::cm::ManagedService, cppmicroservices::cmimpl::TrackedServiceWrapper<cppmicroservices::service::cm::ManagedService> >::TypeTraits>::*(std::shared_ptr<cppmicroservices::detail::TrackedService<cppmicroservices::service::cm::ManagedService, cppmicroservices::ServiceTrackerCustomizer<cppmicroservices::service::cm::ManagedService, cppmicroservices::cmimpl::TrackedServiceWrapper<cppmicroservices::service::cm::ManagedService> >::TypeTraits> >, std::_Placeholder<1>))(cppmicroservices::ServiceEvent const&)>&, cppmicroservices::ServiceEvent const&)(struct std::_Bind<void (cppmicroservices::detail::TrackedService<cppmicroservices::service::cm::ManagedService, cppmicroservices::ServiceTrackerCustomizer<cppmicroservices::service::cm::ManagedService, cppmicroservices::cmimpl::TrackedServiceWrapper<cppmicroservices::service::cm::ManagedService> >::TypeTraits>::*(std::shared_ptr<cppmicroservices::detail::TrackedService<cppmicroservices::service::cm::ManagedService, cppmicroservices::ServiceTrackerCustomizer<cppmicroservices::service::cm::ManagedService, cppmicroservices::cmimpl::TrackedServiceWrapper<cppmicroservices::service::cm::ManagedService> >::TypeTraits> >, std::_Placeholder<1>))(const cppmicroservices::ServiceEvent&)> & __fn) (/usr/include/c++/10/bits/invoke.h:110)
libConfigurationAdmind.so!std::_Function_handler<void (cppmicroservices::ServiceEvent const&), std::_Bind<void (cppmicroservices::detail::TrackedService<cppmicroservices::service::cm::ManagedService, cppmicroservices::ServiceTrackerCustomizer<cppmicroservices::service::cm::ManagedService, cppmicroservices::cmimpl::TrackedServiceWrapper<cppmicroservices::service::cm::ManagedService> >::TypeTraits>::*(std::shared_ptr<cppmicroservices::detail::TrackedService<cppmicroservices::service::cm::ManagedService, cppmicroservices::ServiceTrackerCustomizer<cppmicroservices::service::cm::ManagedService, cppmicroservices::cmimpl::TrackedServiceWrapper<cppmicroservices::service::cm::ManagedService> >::TypeTraits> >, std::_Placeholder<1>))(cppmicroservices::ServiceEvent const&)> >::_M_invoke(std::_Any_data const&, cppmicroservices::ServiceEvent const&)(const union std::_Any_data & __functor, const class cppmicroservices::ServiceEvent & __args#0) (/usr/include/c++/10/bits/std_function.h:291)
libCppMicroServicesd.so.3.7.6!std::function<void (cppmicroservices::ServiceEvent const&)>::operator()(cppmicroservices::ServiceEvent const&) const(const class std::function<void(const cppmicroservices::ServiceEvent&)> * const this, const class cppmicroservices::ServiceEvent & __args#0) (/usr/include/c++/10/bits/std_function.h:622)
libCppMicroServicesd.so.3.7.6!cppmicroservices::ServiceListenerEntry::CallDelegate(const class cppmicroservices::ServiceListenerEntry * const this, const class cppmicroservices::ServiceEvent & event) (/home/tcormack/dev/AsyncWorkDeadlock/CppMicroServices/framework/src/service/ServiceListenerEntry.cpp:143)
libCppMicroServicesd.so.3.7.6!cppmicroservices::ServiceListeners::ServiceChanged(cppmicroservices::ServiceListeners * const this, cppmicroservices::ServiceListeners::ServiceListenerEntries & receivers, const cppmicroservices::ServiceEvent & evt, cppmicroservices::ServiceListeners::ServiceListenerEntries & matchBefore) (/home/tcormack/dev/AsyncWorkDeadlock/CppMicroServices/framework/src/service/ServiceListeners.cpp:434)
libCppMicroServicesd.so.3.7.6!cppmicroservices::ServiceListeners::ServiceChanged(cppmicroservices::ServiceListeners * const this, cppmicroservices::ServiceListeners::ServiceListenerEntries & receivers, const cppmicroservices::ServiceEvent & evt) (/home/tcormack/dev/AsyncWorkDeadlock/CppMicroServices/framework/src/service/ServiceListeners.cpp:412)
libCppMicroServicesd.so.3.7.6!cppmicroservices::ServiceRegistry::RegisterService(cppmicroservices::ServiceRegistry * const this, cppmicroservices::BundlePrivate * bundle, const cppmicroservices::InterfaceMapConstPtr & service, const cppmicroservices::ServiceProperties & properties) (/home/tcormack/dev/AsyncWorkDeadlock/CppMicroServices/framework/src/service/ServiceRegistry.cpp:132)
libCppMicroServicesd.so.3.7.6!cppmicroservices::BundleContext::RegisterService(cppmicroservices::BundleContext * const this, const cppmicroservices::InterfaceMapConstPtr & service, const cppmicroservices::ServiceProperties & properties) (/home/tcormack/dev/AsyncWorkDeadlock/CppMicroServices/framework/src/bundle/BundleContext.cpp:197)
libDeclarativeServicesd.so!cppmicroservices::scrimpl::RegistrationManager::RegisterService(cppmicroservices::scrimpl::RegistrationManager * const this, const std::shared_ptr<cppmicroservices::ServiceFactory> & factory, const cppmicroservices::ServiceProperties & props) (/home/tcormack/dev/AsyncWorkDeadlock/CppMicroServices/compendium/DeclarativeServices/src/manager/RegistrationManager.cpp:100)
libDeclarativeServicesd.so!cppmicroservices::scrimpl::ComponentConfigurationImpl::RegisterService(cppmicroservices::scrimpl::ComponentConfigurationImpl * const this) (/home/tcormack/dev/AsyncWorkDeadlock/CppMicroServices/compendium/DeclarativeServices/src/manager/ComponentConfigurationImpl.cpp:340)
libDeclarativeServicesd.so!cppmicroservices::scrimpl::CCUnsatisfiedReferenceState::Register(cppmicroservices::scrimpl::CCUnsatisfiedReferenceState * const this, cppmicroservices::scrimpl::ComponentConfigurationImpl & mgr) (/home/tcormack/dev/AsyncWorkDeadlock/CppMicroServices/compendium/DeclarativeServices/src/manager/states/CCUnsatisfiedReferenceState.cpp:58)
libDeclarativeServicesd.so!cppmicroservices::scrimpl::ComponentConfigurationImpl::Register(cppmicroservices::scrimpl::ComponentConfigurationImpl * const this) (/home/tcormack/dev/AsyncWorkDeadlock/CppMicroServices/compendium/DeclarativeServices/src/manager/ComponentConfigurationImpl.cpp:428)
libDeclarativeServicesd.so!cppmicroservices::scrimpl::ComponentConfigurationImpl::ConfigChangedState(cppmicroservices::scrimpl::ComponentConfigurationImpl * const this, const cppmicroservices::scrimpl::ConfigChangeNotification & notification) (/home/tcormack/dev/AsyncWorkDeadlock/CppMicroServices/compendium/DeclarativeServices/src/manager/ComponentConfigurationImpl.cpp:301)
libDeclarativeServicesd.so!std::__invoke_impl<void, void (cppmicroservices::scrimpl::ComponentConfigurationImpl::*&)(cppmicroservices::scrimpl::ConfigChangeNotification const&), cppmicroservices::scrimpl::ComponentConfigurationImpl*&, cppmicroservices::scrimpl::ConfigChangeNotification const&>(void (cppmicroservices::scrimpl::ComponentConfigurationImpl::*&)(cppmicroservices::scrimpl::ComponentConfigurationImpl * const, const cppmicroservices::scrimpl::ConfigChangeNotification &) __f, cppmicroservices::scrimpl::ComponentConfigurationImpl *& __t) (/usr/include/c++/10/bits/invoke.h:73)
libDeclarativeServicesd.so!std::__invoke<void (cppmicroservices::scrimpl::ComponentConfigurationImpl::*&)(cppmicroservices::scrimpl::ConfigChangeNotification const&), cppmicroservices::scrimpl::ComponentConfigurationImpl*&, cppmicroservices::scrimpl::ConfigChangeNotification const&>(void (cppmicroservices::scrimpl::ComponentConfigurationImpl::*&)(cppmicroservices::scrimpl::ComponentConfigurationImpl * const, const cppmicroservices::scrimpl::ConfigChangeNotification &) __fn) (/usr/include/c++/10/bits/invoke.h:95)
libDeclarativeServicesd.so!std::_Bind<void (cppmicroservices::scrimpl::ComponentConfigurationImpl::*(cppmicroservices::scrimpl::ComponentConfigurationImpl*, std::_Placeholder<1>))(cppmicroservices::scrimpl::ConfigChangeNotification const&)>::__call<void, cppmicroservices::scrimpl::ConfigChangeNotification const&, 0ul, 1ul>(std::tuple<cppmicroservices::scrimpl::ConfigChangeNotification const&>&&, std::_Index_tuple<0ul, 1ul>)(std::_Bind<void (cppmicroservices::scrimpl::ComponentConfigurationImpl::*(cppmicroservices::scrimpl::ComponentConfigurationImpl*, std::_Placeholder<1>))(const cppmicroservices::scrimpl::ConfigChangeNotification&)> * const this, std::tuple<cppmicroservices::scrimpl::ConfigChangeNotification const&> && __args) (/usr/include/c++/10/functional:416)
libDeclarativeServicesd.so!std::_Bind<void (cppmicroservices::scrimpl::ComponentConfigurationImpl::*(cppmicroservices::scrimpl::ComponentConfigurationImpl*, std::_Placeholder<1>))(cppmicroservices::scrimpl::ConfigChangeNotification const&)>::operator()<cppmicroservices::scrimpl::ConfigChangeNotification const&, void>(cppmicroservices::scrimpl::ConfigChangeNotification const&)(std::_Bind<void (cppmicroservices::scrimpl::ComponentConfigurationImpl::*(cppmicroservices::scrimpl::ComponentConfigurationImpl*, std::_Placeholder<1>))(const cppmicroservices::scrimpl::ConfigChangeNotification&)> * const this) (/usr/include/c++/10/functional:499)
libDeclarativeServicesd.so!std::__invoke_impl<void, std::_Bind<void (cppmicroservices::scrimpl::ComponentConfigurationImpl::*(cppmicroservices::scrimpl::ComponentConfigurationImpl*, std::_Placeholder<1>))(cppmicroservices::scrimpl::ConfigChangeNotification const&)>&, cppmicroservices::scrimpl::ConfigChangeNotification const&>(std::__invoke_other, std::_Bind<void (cppmicroservices::scrimpl::ComponentConfigurationImpl::*(cppmicroservices::scrimpl::ComponentConfigurationImpl*, std::_Placeholder<1>))(cppmicroservices::scrimpl::ConfigChangeNotification const&)>&, cppmicroservices::scrimpl::ConfigChangeNotification const&)(std::_Bind<void (cppmicroservices::scrimpl::ComponentConfigurationImpl::*(cppmicroservices::scrimpl::ComponentConfigurationImpl*, std::_Placeholder<1>))(const cppmicroservices::scrimpl::ConfigChangeNotification&)> & __f) (/usr/include/c++/10/bits/invoke.h:60)
libDeclarativeServicesd.so!std::__invoke_r<void, std::_Bind<void (cppmicroservices::scrimpl::ComponentConfigurationImpl::*(cppmicroservices::scrimpl::ComponentConfigurationImpl*, std::_Placeholder<1>))(cppmicroservices::scrimpl::ConfigChangeNotification const&)>&, cppmicroservices::scrimpl::ConfigChangeNotification const&>(std::_Bind<void (cppmicroservices::scrimpl::ComponentConfigurationImpl::*(cppmicroservices::scrimpl::ComponentConfigurationImpl*, std::_Placeholder<1>))(cppmicroservices::scrimpl::ConfigChangeNotification const&)>&, cppmicroservices::scrimpl::ConfigChangeNotification const&)(std::_Bind<void (cppmicroservices::scrimpl::ComponentConfigurationImpl::*(cppmicroservices::scrimpl::ComponentConfigurationImpl*, std::_Placeholder<1>))(const cppmicroservices::scrimpl::ConfigChangeNotification&)> & __fn) (/usr/include/c++/10/bits/invoke.h:110)
libDeclarativeServicesd.so!std::_Function_handler<void (cppmicroservices::scrimpl::ConfigChangeNotification const&), std::_Bind<void (cppmicroservices::scrimpl::ComponentConfigurationImpl::*(cppmicroservices::scrimpl::ComponentConfigurationImpl*, std::_Placeholder<1>))(cppmicroservices::scrimpl::ConfigChangeNotification const&)> >::_M_invoke(std::_Any_data const&, cppmicroservices::scrimpl::ConfigChangeNotification const&)(const std::_Any_data & __functor, const cppmicroservices::scrimpl::ConfigChangeNotification & __args#0) (/usr/include/c++/10/bits/std_function.h:291)
libDeclarativeServicesd.so!std::function<void (cppmicroservices::scrimpl::ConfigChangeNotification const&)>::operator()(cppmicroservices::scrimpl::ConfigChangeNotification const&) const(const std::function<void(const cppmicroservices::scrimpl::ConfigChangeNotification&)> * const this, const cppmicroservices::scrimpl::ConfigChangeNotification & __args#0) (/usr/include/c++/10/bits/std_function.h:622)
libDeclarativeServicesd.so!cppmicroservices::scrimpl::ConfigurationNotifier::NotifyAllListeners(cppmicroservices::scrimpl::ConfigurationNotifier * const this, const std::string & pid, cppmicroservices::service::cm::ConfigurationEventType type, std::shared_ptr<cppmicroservices::AnyMap> properties) (/home/tcormack/dev/AsyncWorkDeadlock/CppMicroServices/compendium/DeclarativeServices/src/manager/ConfigurationNotifier.cpp:255)
libDeclarativeServicesd.so!cppmicroservices::service::cm::ConfigurationListenerImpl::configurationEvent(cppmicroservices::service::cm::ConfigurationListenerImpl * const this, const cppmicroservices::service::cm::ConfigurationEvent & event) (/home/tcormack/dev/AsyncWorkDeadlock/CppMicroServices/compendium/DeclarativeServices/src/ConfigurationListenerImpl.cpp:93)
libConfigurationAdmind.so!operator()(const struct {...} * const __closure) (/home/tcormack/dev/AsyncWorkDeadlock/CppMicroServices/compendium/ConfigurationAdmin/src/ConfigurationAdminImpl.cpp:676)
libConfigurationAdmind.so!operator()(cppmicroservices::cmimpl::ConfigurationAdminImpl * this) (/home/tcormack/dev/AsyncWorkDeadlock/CppMicroServices/compendium/ConfigurationAdmin/src/ConfigurationAdminImpl.cpp:1059)
libConfigurationAdmind.so!std::__invoke_impl<void, cppmicroservices::cmimpl::ConfigurationAdminImpl::PerformAsync<cppmicroservices::cmimpl::ConfigurationAdminImpl::NotifyConfigurationUpdated(const string&, long unsigned int)::<lambda()> >::<lambda()>&>(std::__invoke_other, struct {...} &)(struct {...} & __f) (/usr/include/c++/10/bits/invoke.h:60)
libConfigurationAdmind.so!std::__invoke_r<void, cppmicroservices::cmimpl::ConfigurationAdminImpl::PerformAsync<cppmicroservices::cmimpl::ConfigurationAdminImpl::NotifyConfigurationUpdated(const string&, long unsigned int)::<lambda()> >::<lambda()>&>(struct {...} &)(struct {...} & __fn) (/usr/include/c++/10/bits/invoke.h:110)
libConfigurationAdmind.so!operator()(std::__future_base::_Task_state<cppmicroservices::cmimpl::ConfigurationAdminImpl::PerformAsync<cppmicroservices::cmimpl::ConfigurationAdminImpl::NotifyConfigurationUpdated(const string&, long unsigned int)::<lambda()> >::<lambda()>, std::allocator<int>, void()> * const this) (/usr/include/c++/10/future:1432)
libConfigurationAdmind.so!std::__future_base::_Task_setter<std::unique_ptr<std::__future_base::_Result<void>, std::__future_base::_Result_base::_Deleter>, std::__future_base::_Task_state<_Fn, _Alloc, _Res(_Args ...)>::_M_run<cppmicroservices::cmimpl::ConfigurationAdminImpl::PerformAsync<cppmicroservices::cmimpl::ConfigurationAdminImpl::NotifyConfigurationUpdated(const string&, long unsigned int)::<lambda()> >::<lambda()>, std::allocator<int>, void, {}>::<lambda()>, void>::operator()(void) const(const std::__future_base::_Task_setter<std::unique_ptr<std::__future_base::_Result<void>, std::__future_base::_Result_base::_Deleter>, std::__future_base::_Task_state<_Fn, _Alloc, _Res(_Args ...)>::_M_run<cppmicroservices::cmimpl::ConfigurationAdminImpl::PerformAsync<cppmicroservices::cmimpl::ConfigurationAdminImpl::NotifyConfigurationUpdated(const string&, long unsigned int)::<lambda()> >::<lambda()>, std::allocator<int>, void, {}>::<lambda()>, void> * const this) (/usr/include/c++/10/future:1373)
libConfigurationAdmind.so!std::__invoke_impl<std::unique_ptr<std::__future_base::_Result<void>, std::__future_base::_Result_base::_Deleter>, std::__future_base::_Task_setter<std::unique_ptr<std::__future_base::_Result<void>, std::__future_base::_Result_base::_Deleter>, std::__future_base::_Task_state<_Fn, _Alloc, _Res(_Args ...)>::_M_run<cppmicroservices::cmimpl::ConfigurationAdminImpl::PerformAsync<cppmicroservices::cmimpl::ConfigurationAdminImpl::NotifyConfigurationUpdated(const string&, long unsigned int)::<lambda()> >::<lambda()>, std::allocator<int>, void, {}>::<lambda()>, void>&>(std::__invoke_other, std::__future_base::_Task_setter<std::unique_ptr<std::__future_base::_Result<void>, std::__future_base::_Result_base::_Deleter>, std::__future_base::_Task_state<_Fn, _Alloc, _Res(_Args ...)>::_M_run<cppmicroservices::cmimpl::ConfigurationAdminImpl::PerformAsync<cppmicroservices::cmimpl::ConfigurationAdminImpl::NotifyConfigurationUpdated(const string&, long unsigned int)::<lambda()> >::<lambda()>, std::allocator<int>, void, {}>::<lambda()>, void> &)(std::__future_base::_Task_setter<std::unique_ptr<std::__future_base::_Result<void>, std::__future_base::_Result_base::_Deleter>, std::__future_base::_Task_state<_Fn, _Alloc, _Res(_Args ...)>::_M_run<cppmicroservices::cmimpl::ConfigurationAdminImpl::PerformAsync<cppmicroservices::cmimpl::ConfigurationAdminImpl::NotifyConfigurationUpdated(const string&, long unsigned int)::<lambda()> >::<lambda()>, std::allocator<int>, void, {}>::<lambda()>, void> & __f) (/usr/include/c++/10/bits/invoke.h:60)
libConfigurationAdmind.so!std::__invoke_r<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter>, std::__future_base::_Task_setter<std::unique_ptr<std::__future_base::_Result<void>, std::__future_base::_Result_base::_Deleter>, std::__future_base::_Task_state<_Fn, _Alloc, _Res(_Args ...)>::_M_run<cppmicroservices::cmimpl::ConfigurationAdminImpl::PerformAsync<cppmicroservices::cmimpl::ConfigurationAdminImpl::NotifyConfigurationUpdated(const string&, long unsigned int)::<lambda()> >::<lambda()>, std::allocator<int>, void, {}>::<lambda()>, void>&>(std::__future_base::_Task_setter<std::unique_ptr<std::__future_base::_Result<void>, std::__future_base::_Result_base::_Deleter>, std::__future_base::_Task_state<_Fn, _Alloc, _Res(_Args ...)>::_M_run<cppmicroservices::cmimpl::ConfigurationAdminImpl::PerformAsync<cppmicroservices::cmimpl::ConfigurationAdminImpl::NotifyConfigurationUpdated(const string&, long unsigned int)::<lambda()> >::<lambda()>, std::allocator<int>, void, {}>::<lambda()>, void> &)(std::__future_base::_Task_setter<std::unique_ptr<std::__future_base::_Result<void>, std::__future_base::_Result_base::_Deleter>, std::__future_base::_Task_state<_Fn, _Alloc, _Res(_Args ...)>::_M_run<cppmicroservices::cmimpl::ConfigurationAdminImpl::PerformAsync<cppmicroservices::cmimpl::ConfigurationAdminImpl::NotifyConfigurationUpdated(const string&, long unsigned int)::<lambda()> >::<lambda()>, std::allocator<int>, void, {}>::<lambda()>, void> & __fn) (/usr/include/c++/10/bits/invoke.h:113)
libConfigurationAdmind.so!std::_Function_handler<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter>(), std::__future_base::_Task_setter<std::unique_ptr<std::__future_base::_Result<void>, std::__future_base::_Result_base::_Deleter>, std::__future_base::_Task_state<_Fn, _Alloc, _Res(_Args ...)>::_M_run<cppmicroservices::cmimpl::ConfigurationAdminImpl::PerformAsync<cppmicroservices::cmimpl::ConfigurationAdminImpl::NotifyConfigurationUpdated(const string&, long unsigned int)::<lambda()> >::<lambda()>, std::allocator<int>, void, {}>::<lambda()>, void> >::_M_invoke(const std::_Any_data &)(const std::_Any_data & __functor) (/usr/include/c++/10/bits/std_function.h:292)
std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>::operator()() const(const std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter>()> * const this) (/usr/include/c++/10/bits/std_function.h:622)
libConfigurationAdmind.so!std::__future_base::_State_baseV2::_M_do_set(std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>*, bool*)(std::__future_base::_State_baseV2 * const this, std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter>()> * __f, bool * __did_set) (/usr/include/c++/10/future:572)
std::__invoke_impl<void, void (std::__future_base::_State_baseV2::*)(std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>*, bool*), std::__future_base::_State_baseV2*, std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>*, bool*>(std::__invoke_memfun_deref, void (std::__future_base::_State_baseV2::*&&)(std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>*, bool*), std::__future_base::_State_baseV2*&&, std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>*&&, bool*&&)(void (std::__future_base::_State_baseV2::*&&)(std::__future_base::_State_baseV2 * const, std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter>()> *, bool *) __f, std::__future_base::_State_baseV2 *&& __t) (/usr/include/c++/10/bits/invoke.h:73)
std::__invoke<void (std::__future_base::_State_baseV2::*)(std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>*, bool*), std::__future_base::_State_baseV2*, std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>*, bool*>(void (std::__future_base::_State_baseV2::*&&)(std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>*, bool*), std::__future_base::_State_baseV2*&&, std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>*&&, bool*&&)(void (std::__future_base::_State_baseV2::*&&)(std::__future_base::_State_baseV2 * const, std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter>()> *, bool *) __fn) (/usr/include/c++/10/bits/invoke.h:95)
std::call_once<void (std::__future_base::_State_baseV2::*)(std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>*, bool*), std::__future_base::_State_baseV2*, std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>*, bool*>(std::once_flag&, void (std::__future_base::_State_baseV2::*&&)(std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>*, bool*), std::__future_base::_State_baseV2*&&, std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>*&&, bool*&&)::{lambda()#1}::operator()() const() (/usr/include/c++/10/mutex:717)
std::call_once<void (std::__future_base::_State_baseV2::*)(std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>*, bool*), std::__future_base::_State_baseV2*, std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>*, bool*>(std::once_flag&, void (std::__future_base::_State_baseV2::*&&)(std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>*, bool*), std::__future_base::_State_baseV2*&&, std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>*&&, bool*&&)::{lambda()#2}::operator()() const(const struct {...} * const this) (/usr/include/c++/10/mutex:722)
std::call_once<void (std::__future_base::_State_baseV2::*)(std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>*, bool*), std::__future_base::_State_baseV2*, std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>*, bool*>(std::once_flag&, void (std::__future_base::_State_baseV2::*&&)(std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>*, bool*), std::__future_base::_State_baseV2*&&, std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>*&&, bool*&&)::{lambda()#2}::_FUN()() (/usr/include/c++/10/mutex:722)
libpthread.so.0!__pthread_once_slow(pthread_once_t * once_control, void (*)(void) init_routine) (pthread_once.c:116)
__gthread_once(__gthread_once_t * __once, void (*)(void) __func) (/usr/include/x86_64-linux-gnu/c++/10/bits/gthr-default.h:700)
std::call_once<void (std::__future_base::_State_baseV2::*)(std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>*, bool*), std::__future_base::_State_baseV2*, std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>*, bool*>(std::once_flag&, void (std::__future_base::_State_baseV2::*&&)(std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>*, bool*), std::__future_base::_State_baseV2*&&, std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>*&&, bool*&&)(std::once_flag & __once, void (std::__future_base::_State_baseV2::*&&)(std::__future_base::_State_baseV2 * const, std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter>()> *, bool *) __f) (/usr/include/c++/10/mutex:729)
libConfigurationAdmind.so!std::__future_base::_State_baseV2::_M_set_result(std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter> ()>, bool)(std::__future_base::_State_baseV2 * const this, std::function<std::unique_ptr<std::__future_base::_Result_base, std::__future_base::_Result_base::_Deleter>()> __res, bool __ignore_failure) (/usr/include/c++/10/future:412)
libConfigurationAdmind.so!std::__future_base::_Task_state<cppmicroservices::cmimpl::ConfigurationAdminImpl::PerformAsync<cppmicroservices::cmimpl::ConfigurationAdminImpl::NotifyConfigurationUpdated(const string&, long unsigned int)::<lambda()> >::<lambda()>, std::allocator<int>, void()>::_M_run(void)(std::__future_base::_Task_state<cppmicroservices::cmimpl::ConfigurationAdminImpl::PerformAsync<cppmicroservices::cmimpl::ConfigurationAdminImpl::NotifyConfigurationUpdated(const string&, long unsigned int)::<lambda()> >::<lambda()>, std::allocator<int>, void()> * const this) (/usr/include/c++/10/future:1435)
std::packaged_task<void ()>::operator()()(std::packaged_task<void()> * const this) (/usr/include/c++/10/future:1568)
test::AsyncWorkServiceInline::post(std::packaged_task<void ()>&&)(test::AsyncWorkServiceInline * const this, std::packaged_task<void()> && task) (/home/tcormack/dev/AsyncWorkDeadlock/CppMicroServices/compendium/DeclarativeServices/test/gtest/TestAsyncWorkService.cpp:91)
libConfigurationAdmind.so!cppmicroservices::cmimpl::CMAsyncWorkService::post(std::packaged_task<void ()>&&)(cppmicroservices::cmimpl::CMAsyncWorkService * const this, std::packaged_task<void()> && task) (/home/tcormack/dev/AsyncWorkDeadlock/CppMicroServices/compendium/ConfigurationAdmin/src/CMAsyncWorkService.cpp:190)
libConfigurationAdmind.so!cppmicroservices::cmimpl::ConfigurationAdminImpl::PerformAsync<cppmicroservices::cmimpl::ConfigurationAdminImpl::NotifyConfigurationUpdated(const string&, long unsigned int)::<lambda()> >(struct {...} &&)(cppmicroservices::cmimpl::ConfigurationAdminImpl * const this, struct {...} && f) (/home/tcormack/dev/AsyncWorkDeadlock/CppMicroServices/compendium/ConfigurationAdmin/src/ConfigurationAdminImpl.cpp:1065)
libConfigurationAdmind.so!cppmicroservices::cmimpl::ConfigurationAdminImpl::NotifyConfigurationUpdated(cppmicroservices::cmimpl::ConfigurationAdminImpl * const this, const std::string & pid, const unsigned long changeCount) (/home/tcormack/dev/AsyncWorkDeadlock/CppMicroServices/compendium/ConfigurationAdmin/src/ConfigurationAdminImpl.cpp:740)
libConfigurationAdmind.so!cppmicroservices::cmimpl::ConfigurationImpl::Update(cppmicroservices::cmimpl::ConfigurationImpl * const this, cppmicroservices::AnyMap newProperties) (/home/tcormack/dev/AsyncWorkDeadlock/CppMicroServices/compendium/ConfigurationAdmin/src/ConfigurationImpl.cpp:118)
test::TestAsyncWorkServiceEndToEnd_testAsyncWorkServiceDeadlock_Test::TestBody(test::TestAsyncWorkServiceEndToEnd_testAsyncWorkServiceDeadlock_Test * const this) (/home/tcormack/dev/AsyncWorkDeadlock/CppMicroServices/compendium/DeclarativeServices/test/gtest/TestAsyncWorkService.cpp:362)
libgtestd.so.1.14.0!testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test, void>(testing::Test * object, void (testing::Test::*)(testing::Test * const) method, const char * location) (/home/tcormack/dev/AsyncWorkDeadlock/CppMicroServices/third_party/googletest/googletest/src/gtest.cc:2612)
libgtestd.so.1.14.0!testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>(testing::Test * object, void (testing::Test::*)(testing::Test * const) method, const char * location) (/home/tcormack/dev/AsyncWorkDeadlock/CppMicroServices/third_party/googletest/googletest/src/gtest.cc:2648)
libgtestd.so.1.14.0!testing::Test::Run(testing::Test * const this) (/home/tcormack/dev/AsyncWorkDeadlock/CppMicroServices/third_party/googletest/googletest/src/gtest.cc:2687)
libgtestd.so.1.14.0!testing::TestInfo::Run(testing::TestInfo * const this) (/home/tcormack/dev/AsyncWorkDeadlock/CppMicroServices/third_party/googletest/googletest/src/gtest.cc:2836)
libgtestd.so.1.14.0!testing::TestSuite::Run(testing::TestSuite * const this) (/home/tcormack/dev/AsyncWorkDeadlock/CppMicroServices/third_party/googletest/googletest/src/gtest.cc:3015)
libgtestd.so.1.14.0!testing::internal::UnitTestImpl::RunAllTests(testing::internal::UnitTestImpl * const this) (/home/tcormack/dev/AsyncWorkDeadlock/CppMicroServices/third_party/googletest/googletest/src/gtest.cc:5920)
libgtestd.so.1.14.0!testing::internal::HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl * object, bool (testing::internal::UnitTestImpl::*)(testing::internal::UnitTestImpl * const) method, const char * location) (/home/tcormack/dev/AsyncWorkDeadlock/CppMicroServices/third_party/googletest/googletest/src/gtest.cc:2612)
libgtestd.so.1.14.0!testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>(testing::internal::UnitTestImpl * object, bool (testing::internal::UnitTestImpl::*)(testing::internal::UnitTestImpl * const) method, const char * location) (/home/tcormack/dev/AsyncWorkDeadlock/CppMicroServices/third_party/googletest/googletest/src/gtest.cc:2648)
libgtestd.so.1.14.0!testing::UnitTest::Run(testing::UnitTest * const this) (/home/tcormack/dev/AsyncWorkDeadlock/CppMicroServices/third_party/googletest/googletest/src/gtest.cc:5484)
RUN_ALL_TESTS() (/home/tcormack/dev/AsyncWorkDeadlock/CppMicroServices/third_party/googletest/googletest/include/gtest/gtest.h:2317)
main(int argc, char ** argv) (/home/tcormack/dev/AsyncWorkDeadlock/CppMicroServices/compendium/DeclarativeServices/test/gtest/main.cpp:28)`

Reply:
PS probably easier if I just walk you down the stack in vscode so I can show you function calls